components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipFileDataFormat.java

```diff
@@ -72,6 +73,11 @@ public void marshal(final Exchange exchange, final Object graph, final OutputStr
 
     @Override
     public Object unmarshal(final Exchange exchange, final InputStream inputStream) throws Exception {
+        if (supportEmptyDirectory) {
```

Comment:
Maybe a single line will be more beautiful :)
exchange.getIn().setHeader("unzipEmptyDirectorySupported", isSupportEmptyDirectory());